core/src/main/java/hudson/model/Slave.java

```diff
@@ -488,6 +484,59 @@ public FormValidation doCheckRemoteFS(@QueryParameter String value) throws IOExc
 
             return FormValidation.ok();
         }
+
+        /**
+         * Returns the list of {@link ComputerLauncher} descriptors appropriate to the supplied {@link Slave}.
+         *
+         * @param it the {@link Slave} or {@code null} to assume the slave is of type {@link #clazz}.
+         * @return the filtered list
+         * @since 2.12
+         */
+        @Nonnull
+        public List<Descriptor<ComputerLauncher>> computerLauncherDescriptors(@CheckForNull Slave it) {
```

Comment:
These should be `@Restricted` since they are intended for use from Jelly views only, IIUC.